server/service/src/invoice/outbound_shipment/update/generate.rs

```diff
@@ -74,27 +74,43 @@ fn set_new_status_datetime(
     invoice: &mut InvoiceRow,
     status: &Option<UpdateOutboundShipmentStatus>,
 ) {
-    if let Some(new_invoice_status) = UpdateOutboundShipmentStatus::full_status_option(status) {
-        let current_datetime = Utc::now().naive_utc();
-        let invoice_status_index = invoice.status.index();
-        let new_invoice_status_index = new_invoice_status.index();
-
-        let is_status_update = |status: InvoiceRowStatus| {
-            new_invoice_status_index >= status.index()
-                && invoice_status_index < new_invoice_status_index
-        };
-
-        if is_status_update(InvoiceRowStatus::Allocated) {
+    let new_status = match status {
+        Some(new_status) if new_status.full_status() != invoice.status => new_status,
+        _ => return,
+    };
+
+    let current_datetime = Utc::now().naive_utc();
+
+    // Status sequence for outbound shipment: New, Allocated, Picked, Shipped
+    match (&invoice.status, new_status) {
```

Comment:
> On a side note though, is the reason we potentially update from say InvoiceRowStatus::New to UpdateOutboundShipmentStatus::Shipped because of the deduping in the changelog?

As per Mark, update shipment input allows you to go up to any status from current status